add h5py as a dependency and move non-host deps to run deps

cekees opened the pull request:
<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Fixes unreported issue of missing h5py.

cekees committed:
add h5py as a dependency and move non-host deps to run deps

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

cekees committed:
bump build number

cekees commented:
@conda-forge-admin, please rerender

zhang-alvin commented:
@cekees, looks like triangle.h can't be found

cekees committed:
put triangle back as host dep

The triangle library should't be required for proteus builds anymore,
but all the references from it weren't removed in the 1.8.0 release.
